addons/survey/static/src/js/survey_result.js

```diff
@@ -114,12 +130,33 @@ publicWidget.registry.SurveyResultChart = publicWidget.Widget.extend({
                         self.chartConfig = self._getSectionResultsChartConfig();
                         break;
                 }
-
-                self._loadChart();
+                window.addEventListener("afterprint", self._onAfterPrint.bind(self));
+                window.addEventListener("beforeprint", self._onBeforePrint.bind(self));
+                self.chart = self._loadChart();
             }
         });
     },
 
+    // -------------------------------------------------------------------------
+    // Handlers
+    // -------------------------------------------------------------------------
+
+    /**
+     * Prepare chart for media print
+     * @private
+     */
+    _onBeforePrint: function () {
+        this.chart.resize(this.printWidth, Math.floor(this.printWidth / this.chart.aspectRatio));
```

Comment:
If we need to keep this I have a few more remarks:

- We can keep "printWidth" defined here with a "const" instead of a widget variable as it does not seem used elsewhere
- What is "630"? pixels?
- Why do we use "630" exactly? Does it give the best ratio on A4 paper?
- If the previous point is actually because of printing format, would be nice to explain it in the method docstring

Reply:
Yes, it is a discrete value to ensure graphs fit into a A4 page, portrait. I change the instance var into a const var in the method and add a comment :+1: